record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/CollectionBuilderUtils.java

```diff
@@ -385,8 +387,9 @@ private MethodSpec buildShimMethod(String name, TypeName mainType, Class<?> abst
                 .toList().toArray(new TypeName[0]);
         var extendedParameterizedType = ParameterizedTypeName.get(ClassName.get(abstractType), wildCardTypeArguments);
         return MethodSpec.methodBuilder(name).addAnnotation(generatedRecordBuilderAnnotation)
```

Comment:
The suppress annotation is only needed when there are casts.

```suggestion

        MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder(name);
        if (!useImmutableCollections) {
            methodBuilder.addAnnotation(suppressWarningsAnnotation);
        }
        return methodBuilder.addAnnotation(generatedRecordBuilderAnnotation)
                .addModifiers(Modifier.PRIVATE, Modifier.STATIC).addTypeVariables(Arrays.asList(typeVariables))
                .returns(parameterizedType).addParameter(extendedParameterizedType, "o").addStatement(code).build();
```